agents/diam_it_test.go

```diff
@@ -1469,39 +1469,27 @@ func testDiamItEmulateTerminate(t *testing.T) {
 	if diamConfigDIR == "dispatchers/diamagent" {
 		t.SkipNow()
 	}
-	var result string
-	//add the second charger
-	chargerProfile := &v1.ChargerWithAPIOpts{
-		ChargerProfile: &engine.ChargerProfile{
-			Tenant:       "cgrates.com",
-			ID:           "CustomCharger",
-			RunID:        "CustomCharger",
-			AttributeIDs: []string{"*constant:*req.Category:custom_charger"},
-			Weight:       20,
-		},
+	//add a default charger
+	tpDirPath := t.TempDir()
```

Comment:
Need to check for an option without temporary files which can create leftovers and also unnecessary increase the file descriptors on the server.

Reply:
TempDir method on testing.T is the recommended way to create temporary directories during tests. It is accompanied by the following comment:
> TempDir returns a temporary directory for the test to use. The directory is automatically removed when the test and all its subtests complete. Each subsequent call to t.TempDir returns a unique directory; if the directory creation fails, TempDir terminates the test by calling Fatal.

The above behaves similarly to doing:

```go
if err = os.MkdirAll("/tmp/testTPs/", 0700); err != nil {
		t.Fatal(err)
}
```

and then using either defer or t.Cleanup to ensure the files are deleted at the end of the test

```go
defer func () {
    if err := os.RemoveAll("/tmp/testTPs/"); err != nil {
		t.Error(err)
    }
}()
```